### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by ryan-cahill**

### CODE REVIEW

File: architect.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      main: &api-port 3000
+    liveness_probe:
+      path: /users
+      port: *api-port
+      interval: 10s
+      failure_threshold: 3
+    environment:
+      PORT: *api-port
+      DB_ADDR: ${{ services.api-db.interfaces.postgres.url }}/${{ parameters.api_db_name }}
+      DB_PORT: ${{ services.api-db.interfaces.postgres.port }}
```
Comment (tjhiggins) [inline architect.yml:61]: remove

File: architect.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    interfaces:
+      main: &api-port 3000
+    liveness_probe:
+      path: /users
+      port: *api-port
+      interval: 10s
+      failure_threshold: 3
+    environment:
+      PORT: *api-port
+      DB_ADDR: ${{ services.api-db.interfaces.postgres.url }}/${{ parameters.api_db_name }}
```
Comment (tjhiggins) [inline architect.yml:60]: Specify the db name as the path for the interface. Then you can simplify to:
```ts
DB_ADDR: ${{ services.api-db.interfaces.mysql.url }}
```

File: architect.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    default: root
+  root_db_pass:
+    description: Root password to assign to the generated database
+    default: architect
+
+services:
+  api-db:
+    image: mysql:8
+    interfaces:
+      postgres:
```
Comment (tjhiggins) [inline architect.yml:34]: mysql

**Review by tjhiggins — CHANGES_REQUESTED** [review]

**Comment by ryan-cahill** [comment]
I don't think we need to worry about the license. This example was modified from the official nest examples anyway

**Pull request merged**